tests/profiles-enabled/test_shipments.py

```diff
@@ -1201,8 +1212,21 @@ def test_permission_link(self):
 
         self.set_user(self.user_2)
 
+        # A tier authenticated user(not shipment owner) cannot access
+        # the permission link objects of a shipment not owned
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # An authenticated user without permission link should not have access
         response = self.client.get(shipment_url)
-        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # An authenticated user can only access the list of shipments he owns with a valid permission link
+        # user_2 owns 1 shipment and has the permission link for the other one, total 2 shipments
+        response = self.client.get(f'{url_shipment_list}?permission_link={valid_permission_id_with_exp}')
```

Comment:
@ajhodges from line 1224 to 1229 is a test on `shipments-list` for an authenticated user which return the user's shipments plus the one for which he has the permission link.